server/coordinator/scheduler/scheduler_manager.go

```diff
@@ -110,6 +113,19 @@ func (m *ManagerImpl) Start(ctx context.Context) error {
 				// Get latest cluster snapshot.
 				clusterSnapshot := m.clusterMetadata.GetClusterSnapshot()
 				log.Debug("scheduler manager invoke", zap.String("clusterSnapshot", fmt.Sprintf("%v", clusterSnapshot)))
+
+				// TODO: Perhaps these codes related to deployMode need to be refactored.
+				// If deployMode is turned on, the scheduler will only be scheduled in the non-stable state.
+				if m.deployMode && clusterSnapshot.Topology.ClusterView.State == storage.ClusterStateStable {
+					continue
+				}
+				if clusterSnapshot.Topology.ClusterView.State == storage.ClusterStatePrepare && len(clusterSnapshot.Topology.ShardViewsMapping) == len(clusterSnapshot.Topology.ClusterView.ShardNodes) {
```

Comment:
Shall we add a method on Topology called `IsPrepareFinished` for this line code?